### PR TITLE
Fix possible arg count error

### PR DESCRIPTION
**Pull request by wrefgtzweve**

### CODE REVIEW

File: lua/advdupe2/sv_clipboard.lua [hunk 1]
```diff
@@ -1001,7 +1006,13 @@ local function CreateEntityFromTable(EntTable, Player)
...
 
 			status, valid = pcall(EntityClass.Func, Player, unpack(ArgList, 1, #EntityClass.Args))
-			if not status then ErrorNoHalt(valid) end
+			if not status then
+				if lastEnt:GetClass() == EntTable.Class then
+					print("Advanced Duplicator 2: Error creating entity, removing last created entity", lastEnt)
+					SafeRemoveEntity(lastEnt)
+				end
+				ErrorNoHaltWithStack(valid)
+			end
```
Comment (thegrb93) [inline lua/advdupe2/sv_clipboard.lua:1015]: Move down to line 1064

File: lua/advdupe2/sv_clipboard.lua [hunk 2]
```diff
@@ -905,6 +905,11 @@ local function IsAllowed(Player, Class, EntityClass)
 	return true
 end
 
+local lastEnt
+hook.Add( "OnEntityCreated", "AdvDupe2_GetLastEntityCreated", function( ent )
+	lastEnt = ent
+end )
+
```
Comment (thegrb93) [inline lua/advdupe2/sv_clipboard.lua:912]: Some factory functions create multiple entities so we may need this to be a list entities and have a table.Empty call before each factory call. This hook should also only be created when we start pasting and removed when we stop pasting (you'll need to suffix the hook name with the player index and change lastEnt to Player.AdvDupe2.lastEntList or something).

**Review by thegrb93 — CHANGES_REQUESTED** [review]

File: lua/advdupe2/sv_clipboard.lua (hunk 2 above)
Comment (wrefgtzweve) [reply lua/advdupe2/sv_clipboard.lua]: It'd be near impossible to figure out which entities we created by the function if it's not done in a single tick (unless im missing something) so i came with the idea to add the hook before the call and remove it after it to catch all created ents in that specific call.

File: lua/advdupe2/sv_clipboard.lua [hunk 3]
```diff
@@ -1050,6 +1057,13 @@ local function CreateEntityFromTable(EntTable, Player)
 
 		return valid
 	else
+		if (status == false) then
+			print("Advanced Duplicator 2: Error creating entity, removing last created entities", lastEnt)
```
Comment (thegrb93) [inline lua/advdupe2/sv_clipboard.lua:1061]: lastEnt here doesn't exist anymore.

File: lua/advdupe2/sv_clipboard.lua [hunk 4]
```diff
@@ -1000,8 +1001,14 @@ local function CreateEntityFromTable(EntTable, Player)
...
 
+			hook.Add( "OnEntityCreated", "AdvDupe2_GetLastEntitiesCreated", function( ent )
+				table.insert( CreatedEntities, ent )
+			end )
+
 			status, valid = pcall(EntityClass.Func, Player, unpack(ArgList, 1, #EntityClass.Args))
-			if not status then ErrorNoHalt(valid) end
+
+			hook.Remove( "OnEntityCreated", "AdvDupe2_GetLastEntitiesCreated" )
+			if not status then ErrorNoHaltWithStack(valid) end
```
Comment (thegrb93) [inline lua/advdupe2/sv_clipboard.lua:1011]: This line could be moved where the remove code is.

**Review by thegrb93 — CHANGES_REQUESTED** [review]

**Comment by thegrb93** [comment]
I just need to take a second to test it tonight

**Pull request merged**